src/map/mob.cpp

```diff
@@ -804,10 +804,10 @@ int mob_attack(dumb_ptr<mob_data> md, tick_t tick)
 
     nullpo_retz(md);
 
-    if ((tbl = map_id2bl(md->target_id)) == nullptr)
+    if (!mob_check_attack(md))
         return 0;
 
-    if (!mob_check_attack(md))
+    if ((tbl = map_id2bl(md->target_id)) == nullptr)
```

Comment:
how does inverting 2 checks that both have the same result (return 0) fix the bug? that would be like changing `if (true == true && 1==1)` to `if (1==1 && true==true)`